src/themes/xresources.rs

```diff
@@ -0,0 +1,75 @@
+#![allow(dead_code)]
+use log::*;
+use once_cell::sync::Lazy;
+use regex::Regex;
+use std::{collections::HashMap, env, path::PathBuf};
+
+#[cfg(not(test))]
+fn read_xresources() -> std::io::Result<String> {
+    let home = env::var("HOME").expect("HOME env var was not set?!");
+    let xresources = PathBuf::from(home + "/.Xresources");
+    debug!(".Xresources @ {:?}", xresources);
+    std::fs::read_to_string(xresources)
+}
+
+#[cfg(test)]
+use tests::read_xresources;
+
+static COLOR_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^\s*\*(?<name>[^: ]+)\s*:\s*(?<color>#[a-f0-9]{6}).*$").unwrap());
+
+pub static COLORS: Lazy<Result<HashMap<String, String>, env::VarError>> = Lazy::new(|| {
+    let home = env::var("HOME")?;
+    let xresources = PathBuf::from(home + "/.Xresources");
+    debug!(".Xresources @ {:?}", xresources);
+    if let Ok(content) = read_xresources() {
+        debug!(".Xresources content:\n{}", content);
+        return Ok(HashMap::from_iter(
+            content
+                .lines()
+                .map(|line| {
+                    COLOR_REGEX
+                        .captures(line)
+                        .map(|caps| (caps["name"].to_string(), caps["color"].to_string()))
+                })
+                .flatten(),
+        ));
+    }
+    warn!(".Xresources not found");
+    Ok(HashMap::new())
```

Comment:
I think it is better to just return an error from this closure if reading fails. This way the actual error message would be displayed in the bar, instead of "color '{name}' is not defined in .Xresources", which may become hard to debug.

Reply:
Yes, I see your point. Have a look at the new version, is this what you meant?

Reply:
Yes, basically. Although I think having the error type be `Error` would be better than `String`. You will just have to write `.map_err(Clone::clone)` when accessing it.

Reply:
Ah! Yes, this looks much cleaner, tbh.